src/main.rs

```diff
@@ -216,229 +226,38 @@ fn cmd_pack(input: &Path, out_dir: &Path) -> Result<()> {
-    // Forge a definition file for each device in the .pdsc file.
-    let mut devices = pdsc.devices.0.into_iter().collect::<Vec<_>>();
-    devices.sort_by(|a, b| a.0.cmp(&b.0));
-
-    for (device_name, device) in devices {
-        // Extract the RAM info from the .pdsc file.
-        let ram = get_ram(&device);
-
-        // Extract the flash algorithm, block & sector size and the erased byte value from the ELF binary.
-        let variant_flash_algorithms = device
-            .algorithms
-            .iter()
-            .map(|flash_algorithm| {
-                let algo = if let Some(ref mut archive) = archive {
-                    crate::parser::extract_flash_algo(
-                        archive.by_name(&flash_algorithm.file_name.as_path().to_string_lossy())?,
-                        &flash_algorithm.file_name,
-                        flash_algorithm.default,
-                    )
-                } else {
-                    crate::parser::extract_flash_algo(
-                        std::fs::File::open(input.join(&flash_algorithm.file_name))?,
-                        &flash_algorithm.file_name,
-                        flash_algorithm.default,
-                    )
-                }?;
-
-                Ok(algo)
-            })
-            .filter_map(
-                |flash_algorithm: Result<RawFlashAlgorithm>| match flash_algorithm {
-                    Ok(flash_algorithm) => Some(flash_algorithm),
-                    Err(error) => {
-                        log::warn!("Failed to parse flash algorithm.");
-                        log::warn!("Reason: {:?}", error);
-                        None
-                    }
-                },
-            )
-            .collect::<Vec<_>>();
-
-        // Extract the flash info from the .pdsc file.
-        let mut flash = None;
-        for memory in device.memories.0.values() {
-            if memory.default && memory.access.read && memory.access.execute && !memory.access.write
-            {
-                flash = Some(FlashRegion {
-                    range: memory.start as u32..memory.start as u32 + memory.size as u32,
-                    is_boot_memory: memory.startup,
-                });
-                break;
-            }
-        }
-
-        // Get the core type.
-        let core = if let Processors::Symmetric(processor) = &device.processor {
-            match &processor.core {
-                Core::CortexM0 => "M0",
-                Core::CortexM0Plus => "M0",
-                Core::CortexM4 => "M4",
-                Core::CortexM3 => "M3",
-                Core::CortexM33 => "M33",
-                Core::CortexM7 => "M7",
-                c => {
-                    bail!("Core '{:?}' is not yet supported for target generation.", c);
-                }
-            }
-        } else {
-            log::warn!("Asymmetric cores are not supported yet.");
-            ""
-        };
+/// Handle the pack subcommand. `input` is either the path
```

Comment:
Is this for the *pack* sub command or for the *arm* sub command? The comment says pack, but the function looks like it is for the arm sub command.